raiden/utils/__init__.py

```diff
@@ -101,6 +102,23 @@ def address_decoder(addr: str) -> typing.Address:
     return addr
 
 
+def address_checksum_and_decode(addr: str) -> typing.Address:
+    """ Accepts a string address and turns it into binary.
+
+        Makes sure that the string address provided starts is 0x prefixed and
+        checksummed according to EIP55 specification
+    """
+    if addr[:2] != '0x':
+        raise InvalidAddress('All provided addresses must be 0x prefixed')
+
+    if not is_checksum_address(addr):
+        raise InvalidAddress('Provided addresses must be EIP55 checksummed')
+
+    addr = unhexlify(addr[2:])
```

Comment:
This is what [`to_canonical_address`](https://github.com/ethereum/eth-utils#to_canonical_addressvalue---bytes) does.